### PR TITLE
Feature/add more context to events

### LINKED ISSUE #5167: get the ID of the "hosting" element
I tried to find an answer with searching in the docs and here - without solution.

I would like to read the ID or name of the table-hosting html element.

`<table id="tbl-test" class="table table-striped"></table>`

called with:

`$("#tbl-test").bootstrapTable({ ...`

how can i get the name "tbl-test" in the table events ?

Edit: I really need the way from INSIDE the event fct, not the "$("#tbl-test")" way.

### PR DESCRIPTION
**Pull request by djhvscf**
**Bug fix?**
No

**New Feature?**
Yes

**Resolve an issue?**
Fix #5167 

**Example(s)?**
https://live.bootstrap-table.com/code/djhvscf/3953

### CODE REVIEW

File: src/bootstrap-table.js [hunk 1]
```diff
@@ -1862,11 +1862,12 @@ class BootstrapTable {
 
   trigger (_name, ...args) {
     const name = `${_name}.bs.table`
+    args[args.length] = this
     this.options[BootstrapTable.EVENTS[name]](...args)
```
Comment (wenzhixin) [inline src/bootstrap-table.js:1866]: this.options[BootstrapTable.EVENTS[name]](...[...args, this])

File: src/bootstrap-table.js [hunk 2]
```diff
@@ -1862,11 +1862,12 @@ class BootstrapTable {
...
-    this.$el.trigger($.Event(name), args)
+    this.$el.trigger($.Event(name, { sender: this }), args)
 
     this.options.onAll(name, args)
```
Comment (wenzhixin) [inline src/bootstrap-table.js:1869]: `this.options.onAll(name, [...args, this])`

File: src/bootstrap-table.js (hunk 1 above)
Comment (djhvscf) [reply src/bootstrap-table.js]: Thanks, done.

File: src/bootstrap-table.js (hunk 2 above)
Comment (djhvscf) [reply src/bootstrap-table.js]: Thanks, done.

**Comment by snoopy72** [comment]
downloaded current version tdday (v1.17.1), removed all existing sources and copied the dist files into my web-project.

but: in my event handlers i cannot find the (new and last) element:

`onColumnSwitch: function(field, checked, e) {
   console.log("onColumnSwitch field="+field+" / checkd: "+checked);
   console.log(e);`

Result:

`onColumnSwitch field=name / checkd: false
undefined`

according to the docs, in this method of event handling, the sender should be in the last parameter.

my fault or went something wrong ?

**Comment by UtechtDustin** [comment]
@snoopy72 do you build the project ?
The dists files in the develop branch (which has this feature) are the version 1.17.1.
So you have have to build the project, use the source files or you have to wait for a new version (we will release a new version soon).

**Comment by snoopy72** [comment]
ah, quick and helpful ... dustin again

maybe i will rebuild today .. if i become too bored ;)
what means "soon" in days - so round about ?

**Comment by UtechtDustin** [comment]
Its planned for the weekend, but we want to merge #5227 first.

**Comment by snoopy72** [comment]
@UtechtDustin THAT is a feature i am waiting very long time for !! i need it for my advanced search which must work with server-requested data (which the current advanced search in BT is not doing). Then i will stop the developing for that feature and will wait for the new version. Seems, it's worth it. Thanks !

**Pull request merged**